viz/cmd/install.go

```diff
@@ -155,3 +164,51 @@ func render(w io.Writer, valuesOverrides map[string]interface{}) error {
 	_, err = w.Write(buf.Bytes())
 	return err
 }
+
+func checkInjectorRunningOrRetryOrExit(retryDeadline time.Duration) {
+	checks := []healthcheck.CategoryID{
+		healthcheck.KubernetesAPIChecks,
+		healthcheck.LinkerdControlPlaneExistenceChecks,
+		healthcheck.LinkerdAPIChecks,
+	}
+
+	hc := healthcheck.NewHealthChecker(checks, &healthcheck.Options{
+		ControlPlaneNamespace: controlPlaneNamespace,
+		KubeConfig:            kubeconfigPath,
+		KubeContext:           kubeContext,
+		Impersonate:           impersonate,
+		ImpersonateGroup:      impersonateGroup,
+		APIAddr:               apiAddr,
+		RetryDeadline:         time.Now().Add(retryDeadline),
+	})
+
+	hc.RunChecks(exitOnError)
+}
+
+func exitOnError(result *healthcheck.CheckResult) {
+	if result.Retry {
+		fmt.Fprintln(os.Stderr, "Waiting for core control plane to become available")
+		return
+	}
+
+	if result.Err != nil && !result.Warning {
+		var msg string
+		switch result.Category {
+		case healthcheck.KubernetesAPIChecks:
+			msg = "Cannot connect to Kubernetes"
+		case healthcheck.LinkerdControlPlaneExistenceChecks:
+			msg = "Cannot find Linkerd"
+		case healthcheck.LinkerdAPIChecks:
+			msg = "Cannot connect to Linkerd"
+		}
+		fmt.Fprintf(os.Stderr, "%s: %s\n", msg, result.Err)
+
+		checkCmd := "linkerd check"
+		if controlPlaneNamespace != defaultLinkerdNamespace {
+			checkCmd += fmt.Sprintf(" --linkerd-namespace %s", controlPlaneNamespace)
+		}
+		fmt.Fprintf(os.Stderr, "Validate the install with: %s\n", checkCmd)
+
+		os.Exit(1)
+	}
+}
```

Comment:
Not a blocker, but seems like something maybe worth putting in `pkg/`? This is the same as what is in `public_api.go`, so worth considering.

Reply:
Yep, Definitely. There are other PRs having this same fn. Will update!